src/emmy/mafs/plot.clj

```diff
@@ -0,0 +1,102 @@
+(ns emmy.mafs.plot
+  (:require [emmy.mafs.core :as mafs]
+            [emmy.expression.compile :as xc]
+            [mentat.clerk-utils.viewers :refer [q]]))
+
+(defn of-x
+  "
+  - `:y`
+  - `:svg-path-props`
+  - `:color`
+  - `:min-sampling-depth`
+  - :`max-sampling-depth`
+  - `:opacity`
+  - `:weight`
+  - `:style`"
+  ([f opts]
+   (of-x (assoc opts :y f)))
+  ([opts]
+   (let [opts (if (map? opts) opts {:y opts})]
+     (mafs/tagged
+      (if (fn? (:y opts))
+        (let [f (xc/compile-fn (:y opts) 1 {:mode :js})]
+          (q
+           (reagent.core/with-let [f' (js/Function. ~@f)]
+             [mafs.plot/OfX ~(assoc opts :y 'f')])))
```

Comment:
it looks like this is the magic moment: if `y` is a fn, then `q` helps us get the body evaluated on the JS side after we compile it, so no quote on `mafs.plot/OfX`, because q is like a "mafs quote"? Otherwise we don't need to bother, Mafs can just be given the form itself to evaluate, so we quote with a tick